### PR TITLE
Add header and pricing banner in the new pricing page.

### PR DESCRIPTION
**Pull request by jkguidaven**
#### Proposed Changes

* Add an optional `title` property to the Jetpack cloud pricing `Header` component to override the existing header text. This allows us to be selective in how we render the header without changing existing components that use it.

* Render `Header` and `ProductStore` in the selector with the new title text.

* Display the `IntroPricingBanner` component in the `ProductStore`. This is mostly copy/paste to the existing implementation in the ProductGrid component. I did not make a re-usable component out of it so as to not complicate the current `ProductGrid` implementation.

#### Testing Instructions

* Boot up this PR
   * Run `git fetch && git checkout add/pricing-page-header-and-banner`
   * Run yarn start-jetpack-cloud
   * Goto http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1
* Or click on Jetpack Cloud live link below
   * Goto /pricing?flags=jetpack/pricing-page-rework-v1
 * Confirm that the Header text is visible with the correct text
 * Confirm that the pricing banner is visible
 
Screen Shot 2022-08-23 at 6 31 13 PM

 * Load the page without the new pricing page by going to http://jetpack.cloud.localhost:3000/pricing or via the Jetpack cloud live link below and goto `/pricing`.
    * Make sure that the header text is correct and did not load the new text.
    * Make sure that the pricing banner is loading properly.

 

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to 1202796695664022-as-1202838239202323

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202838239202323

### CODE REVIEW

File: client/jetpack-cloud/sections/pricing/header/index.tsx
Comment (andrii-lysenko) [inline client/jetpack-cloud/sections/pricing/header/index.tsx]: You can just use `( { title } )` as it is already specified of type Props.

Comment (jkguidaven) [reply client/jetpack-cloud/sections/pricing/header/index.tsx]: Thanks for spotting this Andrii. done on removing it 👍

File: client/my-sites/plans/jetpack-plans/selector.tsx
Comment (andrii-lysenko) [inline client/my-sites/plans/jetpack-plans/selector.tsx]: Hmm, I'm not an expert but I think we try using **context** to specify things like Header, Footer etc. I'd be curious to read some documentation on this matter but haven't found one. 
In the meantime I suggest try to follow current implementation and rather specify title in `client/jetpack-cloud/sections/pricing/controller.tsx` based on Feature Flag.

**Review by andrii-lysenko — COMMENTED** [review]
The changes looks good during testing. There is small adjustment suggested and we need to make sure tests are passing prior to approval.

File: client/my-sites/plans/jetpack-plans/product-store/index.tsx
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/index.tsx]: Can we extract this to a separate component like `IntroBanner`/`intro-banner.tsx`?

File: client/my-sites/plans/jetpack-plans/product-store/style.scss
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/style.scss:54]: What about using this scss?
```suggestion
    
    &__pricing-banner {
        height: 120px;
    
        margin-bottom: 1.25rem;
    
        @include break-large {
            height: 54px;
    
            margin-bottom: 0;
        }
    }
}
```

File: client/my-sites/plans/jetpack-plans/product-store/index.tsx
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/index.tsx]: Why not move this `.map()` call to `useMemo`?

Comment (manzoorwanijk) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: Then you could have

```ts
const productSlugs = useMemo(
	() =>
		[
			...getProductsToDisplay( {
				duration,
				availableProducts,
				purchasedProducts,
				includedInPlanProducts,
			} ),
			...getPlansToDisplay( { duration, currentPlanSlug } ),
		].map( ( { productSlug } ) => productSlug ),
	[ duration, availableProducts, purchasedProducts, includedInPlanProducts, currentPlanSlug ]
);
```

File: client/my-sites/plans/jetpack-plans/selector.tsx
Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/selector.tsx]: Thanks for pointing this out Andrii. This indeed is the right place for declaring the header.

I have move the declaration within the controller context as suggested. 👍

File: client/my-sites/plans/jetpack-plans/product-store/index.tsx
Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: I initially thought of creating a reusable component for this, but I realized that the `allItems` might be used by other components within the `ProductStore` component. I decided to follow how it was implemented in ProductGrid as the safest route. But let me know what you think. I'm also cool with moving this to a separate component. It will definitely declutter some codes in the `ProductStore`.

File: client/my-sites/plans/jetpack-plans/product-store/style.scss
Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/product-store/style.scss:54]: I was hesitating to use nested CSS as I remembered it was not advised in the [CSS coding guidelines](https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/css.md#nesting).

File: client/my-sites/plans/jetpack-plans/product-store/index.tsx
Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: thanks @manzoorwanijk for the suggestion. I agree this one is a lot better.. 👍 💯

File: client/jetpack-cloud/sections/pricing/controller.tsx
Comment (andrii-lysenko) [inline client/jetpack-cloud/sections/pricing/controller.tsx]: This is good, though I suggest either put a comment like `//Default behavior inside component` or simply specify it here `translate( 'Security, performance, and marketing tools made for WordPress' )` to avoid confusion from other folks

**Review by andrii-lysenko — APPROVED** [review]
I'm quite happy with the PR and changes and it works great!
Good from my side and you're good to go after approval from @manzoorwanijk.
LGTM!

File: client/jetpack-cloud/sections/pricing/controller.tsx
Comment (jkguidaven) [reply client/jetpack-cloud/sections/pricing/controller.tsx]: Yeah this make more sense.. I added the default string as suggested.. 👍

File: client/my-sites/plans/jetpack-plans/product-store/index.tsx
Comment (manzoorwanijk) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: That's good thinking James. I was wondering if it would still be better to extract that logic to a custom hook instead. Something like `useProductSlugs()`

Comment (manzoorwanijk) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: The signature of the hook could be something like this:

```ts
function useProductSlugs( { duration, siteId }: SomeType ) {
    //
}
```

Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: That is actually a good idea.. 👍 I'll work on the custom hook..

Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: @manzoorwanijk I am using this signature and just followed how the rest of the custom hooks with multiple parameters are done in the current codebase. 

```
const useProductSlugs = ( siteId: number | null, duration: Duration ): string[] => {
```

Comment (manzoorwanijk) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: Such parameters are not ideal because they can be hard to read at call-sites.
`useProductSlugs( 12345, 'year' )` is not very semantic.

`useProductSlugs( { siteId: 12345, duration: 'year' } )` on the other hand is more readable.

Comment (jkguidaven) [reply client/my-sites/plans/jetpack-plans/product-store/index.tsx]: I added `ProductSlugsProps` type to make the parameter readable  as suggested. 👍

**Review by manzoorwanijk — COMMENTED** [review]
The code looks good now but there is some alignment issue on mobile with those three items:

| Figma | This PR |
|-|-|
|Screenshot 2022-08-24 at 1 45 53 PM|Screenshot 2022-08-24 at 1 46 21 PM|

**Comment by jkguidaven** [comment]
> The code looks good now but there is some alignment issue on mobile with those three items:
> 
> Figma	This PR
> Screenshot 2022-08-24 at 1 45 53 PM	Screenshot 2022-08-24 at 1 46 21 PM

Looks like existing issue w/ the current Pricing banner. I'll look into this.

**Comment by jkguidaven** [comment]
@manzoorwanijk Perhaps we address this in a different PR? The current `IntroPricingBanner` component may need to have its template restructure, and the `ProductGrid` component may need to have its CSS updated.

It's possible that the PR is changing more files than it should. I would strongly suggest to do it in separate PR and fix it entirely from there.

**Comment by manzoorwanijk** [comment]
> 

@jkguidaven I have fixed the issue for you. It looks good now.

**Review by manzoorwanijk — APPROVED** [review]
We can deal with issues on mobile separately. Please ensure to add a task for that.

**Comment by jkguidaven** [comment]
@manzoorwanijk Ticket has been added 1202796695664022-as board 👍

**Comment by manzoorwanijk** [comment]
> Ticket has been added 1202796695664022-as board 👍

Thank you @jkguidaven

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7493549

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7493549).

**Pull request merged**
